shift_detector/precalculations/TextMetadata.py

```diff
@@ -0,0 +1,383 @@
+        stopword_count = 0
+        words = tokenize(text)
+        try:
+            stop = stopwords.words(languages.get(part1=language).name.lower())
+            if (len(words) == 0):
+                return 0.0
+            for word in words:
+                if word.lower() in stop:
+                    stopword_count += 1
+            return stopword_count/ len(words)
+        except OSError as error:
+            raise ValueError('The language ' +
+                             languages.get(part1=self.language).name.lower() +
+                             ' is not supported by StopwordRatioMetadata') from error
+
+
+class DelimiterTypeMetadata(GenericTextMetadata):
+
+    @staticmethod
+    def metadata_name() -> str:
+        return 'delimiter_type'
+
+    def metadata_column_type(self) -> ColumnType:
+        return ColumnType.categorical
+
+    def metadata_function(self, text):
+        for key, value in delimiters.items():
+            if (regex.compile(value).search(text)):
+                return key
+        return 'no delimiter'
+
+
+class NumPartsMetadata(GenericTextMetadata):
+
+    @staticmethod
+    def metadata_name() -> str:
+        return 'num_parts'
+
+    def metadata_column_type(self) -> ColumnType:
+        return ColumnType.numerical
+
+    def metadata_function(self, text):
+        for key, value in delimiters.items():
+            if DelimiterTypeMetadata().metadata_function(text) == key:
+                return len(regex.split(regex.compile(value), text))
+        return 0
+
+
+DetectorFactory.seed = 0  # seed language detection to make it deterministic
+
+
+class LanguagePerParagraph(GenericTextMetadata):
+
+    @staticmethod
+    def metadata_name() -> str:
+        return 'language'
+
+    def metadata_column_type(self) -> ColumnType:
+        return ColumnType.categorical
+
+    @staticmethod
+    def detect_languages(text):
+        if len(text) == 0:
+            detect(text)  # trigger LangDetectException. Throwing one in here smh doesnt work
+        if DelimiterTypeMetadata().metadata_function(text) == 'HTML':
+            parts = re.split(r'<\s*br\s*/?>', text)
+        else:
+            parts = re.split(r'[\n\r]+', text)
+        parts = [x.strip() for x in parts if x.strip()]
+        detected_languages = defaultdict(int)
+        for part in parts:
+            lang = detect(part)
+            detected_languages[lang] += 1
+        return detected_languages
+
+    def metadata_function(self, text):
+        return dictionary_to_sorted_string(self.detect_languages(text))
+
+
+class LanguageMetadata(GenericTextMetadata):
+
+    @staticmethod
+    def metadata_name() -> str:
+        return 'language'
+
+    def metadata_column_type(self) -> ColumnType:
+        return ColumnType.categorical
+
+    def metadata_function(self, text):
+        return detect(text)
+
+
+class ComplexityMetadata(GenericTextMetadata):
```

Comment:
does this complexity metric have a technical term?

Reply:
no, since we use the textstat.text_standard method, that calculates a complexity score based on many scores, there is no tech term. (https://pypi.org/project/textstat/)